And importantly, stuff like this results in failure. I broke one of the tests to ensure that it was doing it correctly.

```diff
>>> list(flatten_nested_lists( [[1,2,3], {"foo": "bar"}] ))  
-[1, 2, 3, {'foo': 'bar'}]  
+[1, 2, 3, {'foo': 'bar'}, 4, 5, 6]
```

So, the written down "expected answer" is not what the code produces. That results in:
```
pytest --doctest-modules --verbose src/esm_parser
=================================================================================================================================== test session starts ===================================================================================================================================
platform darwin -- Python 3.9.5, pytest-7.1.2, pluggy-1.0.0 -- /private/tmp/esm_tools/.direnv/python-3.9.5/bin/python
cachedir: .pytest_cache
rootdir: /private/tmp/esm_tools
collected 2 items

src/esm_parser/esm_parser.py::esm_parser.esm_parser.attach_to_config_and_reduce_keyword PASSED                                                                                                                                                                                      [ 50%]
src/esm_parser/esm_parser.py::esm_parser.esm_parser.flatten_nested_lists FAILED                                                                                                                                                                                                     [100%]

======================================================================================================================================== FAILURES =========================================================================================================================================
__________________________________________________________________________________________________________________ [doctest] esm_parser.esm_parser.flatten_nested_lists ___________________________________________________________________________________________________________________
148     >>> list(flatten_nested_lists( [[1,2,3]] ))
149     [1, 2, 3]
150
151     >>> list(flatten_nested_lists( [1,2,3, [4,5,6], "foo"] ))
152     [1, 2, 3, 4, 5, 6, 'foo']
153
154     >>> list(flatten_nested_lists( [[1,2,3], [4,5,6], [7,8,9]] ))
155     [1, 2, 3, 4, 5, 6, 7, 8, 9]
156
157     >>> list(flatten_nested_lists( [[1,2,3], {"foo": "bar"}] ))
Expected:
    [1, 2, 3, {'foo': 'bar'}, 4, 5, 6]
Got:
    [1, 2, 3, {'foo': 'bar'}]

/private/tmp/esm_tools/src/esm_parser/esm_parser.py:157: DocTestFailure
================================================================================================================================= short test summary info =================================================================================================================================
FAILED src/esm_parser/esm_parser.py::esm_parser.esm_parser.flatten_nested_lists
=============================================================================================================================== 1 failed, 1 passed in 0.61s ===============================================================================================================================

```

--- a/src/esm_parser/esm_parser.py
+++ b/src/esm_parser/esm_parser.py
@@ -1042,31 +1042,18 @@ def add_entry_to_chapter(
 
     # Eg. add_general.mylist -> mylist
     chapter_to_add = add_chapter.split(".")[-1].replace("add_", "")
-    if (
-        chapter_to_add
-        in target_config[model_to_add_to]
-    ):
-        target_config[model_to_add_to][
-            chapter_to_add
-        ] = add_entries
+    if chapter_to_add in target_config[model_to_add_to]:
+        target_config[model_to_add_to][chapter_to_add] = add_entries
     else:
-        if type(
-            target_config[model_to_add_to][
-                chapter_to_add
-            ]
-        ) != type(add_entries):
+        if type(target_config[model_to_add_to][chapter_to_add]) != type(add_entries):
             raise TypeError("Something is wrong")
         else:
             if isinstance(
-                target_config[model_to_add_to][
-                    chapter_to_add
-                ],
+                target_config[model_to_add_to][chapter_to_add],
                 list,
             ):
                 # Define the list to be modified
-                mod_list = target_config[model_to_add_to][
-                    chapter_to_add
-                ]
+                mod_list = target_config[model_to_add_to][chapter_to_add]
                 # Add the entries
                 mod_list.extend(list(flatten_nested_lists(add_entries)))
 
@@ -1078,24 +1065,18 @@ def add_entry_to_chapter(
                             mod_list_no_dupl.append(el)
                     else:
                         mod_list_no_dupl.append(el)
-                target_config[model_to_add_to][
-                    chapter_to_add
-                ] = mod_list_no_dupl
+                target_config[model_to_add_to][chapter_to_add] = mod_list_no_dupl
                 global list_counter
                 list_counter += 1
             elif isinstance(
-                target_config[model_to_add_to][
-                    chapter_to_add
-                ],
+                target_config[model_to_add_to][chapter_to_add],
                 dict,
             ):
                 # If the chapter is a dictionary use dict_merge where the new entries
                 # have priority over the preexisting ones (user choices win over
                 # anything else)
                 dict_merge(
-                    target_config[model_to_add_to][
-                        chapter_to_add
-                    ],
+                    target_config[model_to_add_to][chapter_to_add],
                     add_entries,
                 )
     if list_counter > 1:

--- a/src/esm_parser/esm_parser.py
+++ b/src/esm_parser/esm_parser.py
@@ -1046,7 +1046,9 @@ def add_entry_to_chapter(
         target_config[model_to_add_to][chapter_to_add] = add_entries
     else:
         if type(target_config[model_to_add_to][chapter_to_add]) != type(add_entries):
-            raise TypeError("Something is wrong")
+            error_type = "Type mismatch"
+            error_text = f"Can not add a variable of incompatible type ``{type(add_entries).__name__}`` to ``{chapter_to_add}``"
+            user_error(error_type, error_text)
         else:
             if isinstance(
                 target_config[model_to_add_to][chapter_to_add],

--- a/src/esm_parser/esm_parser.py
+++ b/src/esm_parser/esm_parser.py
@@ -1042,7 +1042,7 @@ def add_entry_to_chapter(
 
     # Eg. add_general.mylist -> mylist
     chapter_to_add = add_chapter.split(".")[-1].replace("add_", "")
-    if chapter_to_add in target_config[model_to_add_to]:
+    if chapter_to_add not in target_config[model_to_add_to]:
         target_config[model_to_add_to][chapter_to_add] = add_entries
     else:
         if type(target_config[model_to_add_to][chapter_to_add]) != type(add_entries):
@@ -1061,7 +1061,7 @@ def add_entry_to_chapter(
                 mod_list_no_dupl = []
                 for el in mod_list:
                     if not isinstance(el, (dict, tuple, list)):
-                        if not el in mod_list_no_dupl:
+                        if el not in mod_list_no_dupl:
                             mod_list_no_dupl.append(el)
                     else:
                         mod_list_no_dupl.append(el)

